source/mesh_deformation/function.cc

```diff
@@ -0,0 +1,124 @@
+/*
+  Copyright (C) 2018 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+ */
+
+
+#include <aspect/mesh_deformation/function.h>
+
+#include <deal.II/numerics/vector_tools.h>
+
+namespace aspect
+{
+  namespace MeshDeformation
+  {
+    template <int dim>
+    Function<dim>::Function()
+      :
+      function(dim)
+    {}
+
+
+
+    template <int dim>
+    void
+    Function<dim>::update ()
+    {
+      function.set_time(this->get_time());
```

Comment:
Just so we do not forget this later on: I think in the final version this should also take into account the parameter `Use years in output instead of seconds`.